src/credentialstore/credentialstore.ts

```diff
@@ -14,10 +18,32 @@ import SqlToolsServerClient from '../languageservice/serviceclient';
  */
 export class CredentialStore implements ICredentialStore {
 
-    constructor(private _client?: SqlToolsServerClient) {
+    private _useNativeCredentials: boolean;
+    private _passwordsMigrated: boolean;
+    private _secretStorage: vscode.SecretStorage;
+
+    constructor(
+        private _context: vscode.ExtensionContext,
+        private _client?: SqlToolsServerClient) {
         if (!this._client) {
             this._client = SqlToolsServerClient.instance;
         }
+        this._secretStorage = this._context.secrets;
+        this._context.globalState.update(Utils.configPasswordsMigrated, false);
+        this._passwordsMigrated = this._context.globalState.get(Utils.configPasswordsMigrated);
```

Comment:
You don't need to store this anymore - just read/set the stored var in initialize directly